### ISSUE TITLE
[UI idea] Implement student specific customizations

### ISSUE THREAD

**Issue by TiBiBa** [Felienne/hedy#1370]
**Idea incl level**
With the implementation of #1317 and the improvements of #1363 we are able to customize classes on adventures, example programs and/or hiding levels. A great start, but we would also allow the customization on a student level instead of a class level. For example, enabling some adventures for specific students who are more advanced and reduce the amount of adventures for other students. This can be combined with additional features such as maximum amount of faulty code runs, amount of programs required to unlock next level. This should be designed and implemented in collaboration with teachers using the Hedy teachers section. Overview of implementation: 

- We keep the original class-level customization
- We can select a student and create a "unique" customization, being prioritized over de class-level one
- New page is implemented `customize-student` to create UI to make changes
- We assume that a student is only in one class, in any other case we simply select the first one 
- We should be able to get an overview of all student-level customizations to enable editing and/or removing
- When removed, the class-level customization should be used again

Not really liking to make the assumption that students are only in one class. They might want to keep their progress or programs from "older" classes or follow two classes at the same time. Not sure how we can best approach this issue. Maybe students can select their current class from a dropdown menu? I also like the idea that students are able to "escape" their class and simply browse the Hedy website without having the need to logout.

**Comment by Felienne** [comment]
> * We keep the original class-level customization
> * We can select a student and create a "unique" customization, being prioritized over de class-level one

It would be fantastic to have the option to do kids specific customization!

> * New page is implemented `customize-student` to create UI to make changes

I am not sure we need a new page? Maybe we can integrate this into the class customization page? Because otherwise you might have to tick all the tick marks again? I can imagine a class list in the existing page where we can select which students this does or does not apply to. I can draw a UI if you do not see how this could look like?

Maybe we also want the option to create subgroups in a class (like "pluskids")

> * We assume that a student is only in one class, in any other case we simply select the first one

> Not really liking to make the assumption that students are only in one class. They might want to keep their progress or programs from "older" classes or follow two classes at the same time. Not sure how we can best approach this issue. Maybe students can select their current class from a dropdown menu? I also like the idea that students are able to "escape" their class and simply browse the Hedy website without having the need to logout.

I understand but I think this is a hypothetical problem that will not occur often, so if we by default pick the final class a kid joined, it should be ok for now. Kids also have a class page on their profile, so we could implement a switcher there? That might be a solution for the subgroups also.

> * We should be able to get an overview of all student-level customizations to enable editing and/or removing
> * When removed, the class-level customization should be used again

Yes (to both)

**Comment by TiBiBa** [comment]
I keep moving this task around as I'm not sure on the correct approach of this issue. Do we still want to pursue student specific customizations? I'm not sure if there is enough added value on this feature so teachers will actually use this. When there are children lacking behind or working ahead of the rest of the group the teacher is always able to create a second "support" or "plus" class to tackle this issue.

**Comment by Felienne** [comment]
Yeah I agree @TiBiBa, I think we can leave this one for now, unless maybe @OnnoEbbens or @MarleenGilsing really want to have it? Maybe we should try to plan a focus group with some teachers to discuss a road map?

**Comment by OnnoEbbens** [comment]
At the moment I don't really need it. I see the advantages of having a students specific customization but I don't have, and maybe never will have, enough time to customize Hedy for individual students.

**Comment by Felienne** [comment]
Closing in favor of a discussion: https://github.com/Felienne/hedy/discussions/3254. Trying to not have issues open that are not really issues (i.e nog really scoped out or not agreed upon we even want it)
